packages/block-editor/src/components/link-control/index.js

```diff
@@ -274,36 +411,76 @@ function LinkControl( {
 
 		return (
 			<div className="block-editor-link-control__search-results-wrapper">
-				{ isInitialSuggestions ? (
-					SearchResultsLabel
-				) : (
-					<VisuallyHidden>{ SearchResultsLabel }</VisuallyHidden>
-				) }
-
+				{ isInitialSuggestions && searchResultsLabel }
 				<div
 					{ ...suggestionsListProps }
 					className={ resultsListClasses }
-					aria-labelledby={ searchResultsLabelId }
+					aria-label={ ariaLabel } // will only be present if there is no visible label
+					aria-labelledby={ searchResultsLabelId } // references the visible label, if there is one
```

Comment:
I'm so very confused about how all of this has evolved. Was this related to the [discussion yesterday in the accessibility channel](https://wordpress.slack.com/archives/C02RP4X03/p1581500184181100)? Unclear to me, since we're not using `for`, but rather inversing it to `aria-labelledby`. I can't imagine that we want _both_ `aria-label` _and_ `aria-labelledby`?

Reply:
Yes exactly. Basically we were too deliberate in our interpretation of the spec. Based on the Slack convo, we should prefer a hidden HTML-based label over `aria-label`. I'm going to amend this.

Reply:
The confusoin was caused by this state from [MDN docs on `listbox`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role):


> **aria-label**
A human-readable string value which identifies the listbox.  If there's a visible label, then aria-labelledby should be used instead to refer to that label.

> **aria-labelledby**
Identifies the visible element or elements in a space-separated list of element IDs which identify the listbox.  If there's no visible label, then aria-label should be used instead to include a label.  (Note: "labelled", with two L's, is the correct spelling based on the accessibility API conventions.)

Note it says if there is no visible label then `aria-label` should be added. If you follow, that's exactly what we were doing when/if the label wasn't visible.

To simplify and conform to what was recommended in Slack I've gone with an HTML tag as the "label" referenced with `aria-labelledby` on the `listbox`.

Reply:
Ah, I think I understand now what was meant. That said, even if it was kept the way it was, if the reason both are applied is because there's a condition by which the visible label would be rendered, then we probably want to include that condition in determining which of these attributes to apply and when. It seems like it might be asking for trouble to leave it to the browser / assistive device to gracefully handle `aria-labelledby` when the corresponding ID'd element may not exist.

(It might also have made it clearer to future contributors who may experience similar confusion as I did)